install/operator/pkg/syndesis/versions/versions.go

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package versions
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/syndesisio/syndesis/install/operator/pkg/util"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	"github.com/go-logr/logr"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/syndesisio/syndesis/install/operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis/install/operator/pkg/apis/syndesis/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type SyndesisApiMigrator interface {
+	// If Syndesis API version is v1alpha1, migrate it to v1beta1
+	Migrate() error
+}
+
+type syndesisApi struct {
+	client           client.Client
+	context          context.Context
+	log              logr.Logger
+	unstructuredApis *unstructured.UnstructuredList
+	v1alpha1         *v1alpha1.Syndesis
+	v1beta1          *v1beta1.Syndesis
+}
+
+// Build and return an SyndesisApiMigrator interface
+func ApiMigrator(c client.Client, ctx context.Context, n string) (r SyndesisApiMigrator, err error) {
+	// Fetch all existing apis in an unstructured list. It is necessary to use an unstructured list
+	// because different apis might have a different structure
+	list := &unstructured.UnstructuredList{
+		Object: map[string]interface{}{
+			"kind":       "Syndesis",
+			"apiVersion": "syndesis.io/v1beta1",
```

Comment:
Also `syndesis.io/v1alpha1` can be queried. I don't know if it changes anything.

Reply:
It doesn't change anything. As it is I can grab both versions: `v1alpha1` and `v1beta1`. Actually after applying the new CRD with both versions, the apiVersion from the `v1alpha1` instances I had created are changed to `v1beta1`, when I do `oc get syndesis`